### ISSUE TITLE
psc-publish: Unable to parse the date for a git tag

### ISSUE THREAD

**Issue by owickstrom** [purescript/purescript#2610]
I have a tag `v0.1.2` in https://github.com/owickstrom/hyper which causes `psc-publish` to fail. I'd normally run it using pulp publish, but it seems to be an issue with psc-publish itself.

```bash
$ psc-publish --version
0.10.5
$ git checkout v0.1.2
...
HEAD is now at c065744... v0.1.2
$ psc-publish
Internal error: this is probably a bug. Please report it:
  https://github.com/purescript/purescript/issues/new

Details:
Unable to parse the date for a git tag: v0.1.2
```

Looking at the tag:

```bash
$ git show v0.1.2
tag v0.1.2
Tagger: Oskar Wickström <oskar.wickstrom@gmail.com>
Date:   Tue Jan 31 07:59:50 2017 +0100

v0.1.2

commit c065744f9b5c895b06566caf13393294ab8fe882
Author: Oskar Wickström <oskar.wickstrom@gmail.com>
Date:   Tue Jan 31 07:59:50 2017 +0100

    v0.1.2
```

The command run at https://github.com/purescript/purescript/blob/62a30ad868c764b36e93a44544325e2eb9c49034/src/Language/PureScript/Publish.hs#L199 seems to be the one that fails. When I run it manually, I get multiline output, which might be the problem (the parsing code seems to expect the first line to be the UTC date):

```bash
$ git show v0.1.2 --no-patch --format='%aI'
tag v0.1.2
Tagger: Oskar Wickström <oskar.wickstrom@gmail.com>

v0.1.2
2017-01-31T07:59:50+01:00
```

OK, if I try `--dry-run` instead:

```bash
$ psc-publish  --dry-run
fatal: ambiguous argument '0.0.0': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
An error occurred, and your package could not be published.
Details:
  While running `git show 0.0.0 --no-patch --format=%aI`:
    readCreateProcess: git "show" "0.0.0" "--no-patch" "--format=%aI" (exit 128):
    failed
```

It tries to show `0.0.0`, which does not exist.

**Comment by hdgarrood** [comment]
Does `psc-publish` normally work on your machine? Are you able to identify anything unique about this tag? Also what version of git are you using?

**Comment by owickstrom** [comment]
It did before, but then I think a ran version 0.10.1 or perhaps 0.10.2 of psc/psc-publish. I get the same error in another project (https://github.com/owickstrom/purescript-spec) by checking out one of its tags (`v0.11.0`) and running `psc-publish --dry-run`.

**Git version**

```bash
$ git --version
git version 2.9.3
```

**Comment by hdgarrood** [comment]
Are you using a version of `psc` from `master`? I don't think the code that runs that command was in 0.10.5; here's the commit that added it: https://github.com/purescript/purescript/commit/9d065d8f4db319af33d3b6feb2b3b91539921e3c

Thanks for the report, now I'll be able to make sure both of these problems are fixed before 0.10.6 :)

**Comment by owickstrom** [comment]
Yes, sorry, I should have checked that. My version is compiled from 220e6ae1797edd461dcdd31f0e851a3530fbbc91.

No problem, I'm glad to contribute to my favorite compiler. 👍

**Comment by hdgarrood** [comment]
Fixed by #2612 and #2613.

**Comment by mcoffin** [comment]
@hdgarrood I'm still seeing this in `0.10.7` in my travis-ci builds [here](https://travis-ci.org/mcoffin/purescript-hot-shots/builds/202034669#L1323).

**Comment by hdgarrood** [comment]
I'm unable to reproduce either of those failures locally. Do they work locally for you? Can you find out what version of git is being used in those builds? Also, can you paste the output of e.g. `git show v0.0.2 --no-patch --format="%aI"` in any environment where it's breaking?

**Comment by hdgarrood** [comment]
(The above command is what `psc-publish` uses to find the git tag time)

**Comment by mcoffin** [comment]
They work locally for me as well, and I've found the root cause of the problem, likely dealing with travis' version of `git`.

Here's an excerpt from my [build log](https://travis-ci.org/mcoffin/purescript-hot-shots#L1311) that I'm using to test:

```
git show $TRAVIS_TAG --no-patch --format=%aI
tag v0.0.2
Tagger: Matt Coffin <mcoffin13@gmail.com>

Version 0.0.2
%aI
```

Notice how it fails to format the date properly.

**Comment by mcoffin** [comment]
I expect that travis' version of git doesn't support the `iso-strict` date type, meaning that the `%aI` placeholder wouldn't exist.

I did, however, find a way to slightly improve the command we're using to call git, so that it only outputs the desired date.

```bash
g tag -l $TRAVIS_TAG '--format=%(taggerdate:iso-strict)'
```

The `iso-strict` date format was added in `git-2.2` (Nov 2014). Travis is still on `git-1.8`. Seems like for the time being the best solution would just be to use a newer version of git by installing it instead of trying to use travis's version.
